src/app/view/components/_common/iframe/iframe-porlet-legacy/iframe-porlet-legacy.component.ts

```diff
@@ -47,21 +57,13 @@ export class IframePortletLegacyComponent implements OnInit {
     }
 
     /**
-     * Handle the iframe load
+     * Handle the custom events emmited by the iframe
      *
      * @param {any} $event
      * @memberof IframePortletLegacyComponent
      */
-    onLoad($event): void {
-        Observable.fromEvent($event.target.contentWindow.document, 'ng-event').subscribe((event: any) => {
-            if (event.detail.name === 'edit-page') {
-                this.dotLoadingIndicatorService.show();
-
-                this.ngZone.run(() => {
-                    this.dotRouterService.goToEditPage(event.detail.data.url);
-                });
-            }
-        });
+    onCustomEvent($event: CustomEvent): void {
+        this.customEventsHandlers[$event.detail.name]($event);
```

Comment:
I think we can do here something like: 

´´´
const customEventHandler = this.customEventsHandlers[$event.detail.name];
if (customEventHandler) {
    customEventHandler($event);
} else {
   custom.next($event)
}
´´´

this way we can have custom event handler in its own componente, maybe can be usefull

Reply:
I don't see how that can be useful, double check the way we use this component.